challenge.rb

```diff
@@ -1,26 +1,34 @@
 
 def capitalize_each_string(input)
   #implement your solution here
+  input.map do |i|
+    i.capitalize
+  end
 end
 
 def fetch_the_dog(input)
   #implement your solution here
+  input.select {|i| i == "dog"}
 end
 
 def no_dogs_allowed(input)
   #implement your solution here
+  input.reject {|i| i == "dog"}
 end
 
 def count_the_animals(input)
   #implement your solution here
+    input.length
```

Comment:
this works.

see: http://batsov.com/articles/2014/02/17/the-elements-of-style-in-ruby-number-13-length-vs-size-vs-count/